### PR TITLE
Added Tabular Format to output and sudomain checking

### PR DESCRIPTION
**Pull request by cle5eland**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs [hunk 1]
```diff
@@ -106,6 +103,28 @@ public override void ExecuteCmdlet()
...
+        private void CheckSubdomainAvailability()
+        {
+            var checkSubdomainInputs = new OperationInputs(this.Subdomain, "IoTApps");
+            var subdomainAvailabilityInfo = this.IotCentralClient.Apps.CheckSubdomainAvailability(checkSubdomainInputs);
+            this.CheckAvailability(subdomainAvailabilityInfo);
+        }
+
+        private void CheckAvailability(AppAvailabilityInfo availabilityInfo)
+        {
+            if (!(availabilityInfo.NameAvailable == true))
```
Comment (emgarten) [inline src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs]: Simplify this to
`if (availabilityInfo.NameAvailable != true)`

File: src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs [hunk 2]
```diff
@@ -106,6 +103,28 @@ public override void ExecuteCmdlet()
...
+        private void CheckNameAvailability()
+        {
+            var checkNameInputs = new OperationInputs(this.Name, "IoTApps");
+            var nameAvailabilityInfo = this.IotCentralClient.Apps.CheckNameAvailability(checkNameInputs);
+            this.CheckAvailability(nameAvailabilityInfo);
+        }
+
+        private void CheckSubdomainAvailability()
+        {
+            var checkSubdomainInputs = new OperationInputs(this.Subdomain, "IoTApps");
```
Comment (emgarten) [inline src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs]: Add a constant for `IoTApps`

File: src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs [hunk 3]
```diff
@@ -106,6 +103,28 @@ public override void ExecuteCmdlet()
...
+            var subdomainAvailabilityInfo = this.IotCentralClient.Apps.CheckSubdomainAvailability(checkSubdomainInputs);
+            this.CheckAvailability(subdomainAvailabilityInfo);
+        }
+
+        private void CheckAvailability(AppAvailabilityInfo availabilityInfo)
```
Comment (emgarten) [inline src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs]: I would rename these from *Check* to something more clear, *EnsureAvailableOrThrow* for all three of these methods would make it easier to tell what will happen

File: src/ResourceManager/IotCentral/Commands.IotCentral/Az.IotCentral.psd1 [hunk 4]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '0.1.0'
```
Comment (emgarten) [inline src/ResourceManager/IotCentral/Commands.IotCentral/Az.IotCentral.psd1]: Is there any reason these module versions are different?

File: src/ResourceManager/IotCentral/Commands.IotCentral/AzureRM.IotCentral.psd1 [hunk 5]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '1.0.0'
+ModuleVersion = '1.0.1'
```
Comment (emgarten) [inline src/ResourceManager/IotCentral/Commands.IotCentral/AzureRM.IotCentral.psd1]: Should this be a minor version bump instead? it isn't just a bug fix.

File: src/ResourceManager/IotCentral/Commands.IotCentral/Az.IotCentral.psd1 (hunk 4 above)
Comment (cle5eland) [reply src/ResourceManager/IotCentral/Commands.IotCentral/Az.IotCentral.psd1]: Differs depending on .NetFramework vs. .NetCore

File: src/ResourceManager/IotCentral/Commands.IotCentral/ChangeLog.md [hunk 6]
```diff
@@ -19,3 +19,9 @@
 -->
 ## Current Release
 * Preview Release
+* Added Tabular Output Format
+* Added Subdomain Checking in New-AzureRmIotCentralApp
+
+## 2018.11.06 - Version 1.0.0
```
Comment (maddieclayton) [inline src/ResourceManager/IotCentral/Commands.IotCentral/ChangeLog.md]: Please only make edits to the ##Current release header

File: src/ResourceManager/IotCentral/Commands.IotCentral/Microsoft.Azure.Commands.IotCentral.generated.format.ps1xml [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+          <TableColumnHeader>
+            <Alignment>Left</Alignment>
+            <Label>Application Id</Label>
+          </TableColumnHeader>
+        </TableHeaders>
+        <TableRowEntries>
+          <TableRowEntry>
+            <TableColumnItems>
+              <TableColumnItem>
+                <Alignment>Left</Alignment>
```
Comment (maddieclayton) [inline src/ResourceManager/IotCentral/Commands.IotCentral/Microsoft.Azure.Commands.IotCentral.generated.format.ps1xml]: Can you really see all of these properties on the regular PowerShell screen? This seems like a lot of properties for table format.

Comment (cle5eland) [reply src/ResourceManager/IotCentral/Commands.IotCentral/Microsoft.Azure.Commands.IotCentral.generated.format.ps1xml]: @maddieclayton they all fit for me, ill check again

Comment (cle5eland) [reply src/ResourceManager/IotCentral/Commands.IotCentral/Microsoft.Azure.Commands.IotCentral.generated.format.ps1xml]: @maddieclayton you're right it doesn't all show on a small powershell, but it autoformats to only show the columns that will fit. is it better for me to reduce the amount of data output all together?

File: src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs (hunk 1 above)
Comment (cle5eland) [reply src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs]: @emgarten addressed all your reviews

File: src/ResourceManager/IotCentral/Commands.IotCentral/Microsoft.Azure.Commands.IotCentral.generated.format.ps1xml (hunk 7 above)
Comment (maddieclayton) [reply src/ResourceManager/IotCentral/Commands.IotCentral/Microsoft.Azure.Commands.IotCentral.generated.format.ps1xml]: @cle5eland As long as the columns that are shown are the ones that you want, that's fine with me.

Comment (cle5eland) [reply src/ResourceManager/IotCentral/Commands.IotCentral/Microsoft.Azure.Commands.IotCentral.generated.format.ps1xml]: Cool, thanks

File: src/ResourceManager/IotCentral/Commands.IotCentral/ChangeLog.md (hunk 6 above)
Comment (cle5eland) [reply src/ResourceManager/IotCentral/Commands.IotCentral/ChangeLog.md]: @maddieclayton i updated I believe, am i doing that right now?

File: src/ResourceManager/IotCentral/Commands.IotCentral/AzureRM.IotCentral.psd1 (hunk 5 above)
Comment (cle5eland) [reply src/ResourceManager/IotCentral/Commands.IotCentral/AzureRM.IotCentral.psd1]: @emgarten resolved with ps team

**Pull request merged**